### PR TITLE
Fixes support for `tryparse` with `ZonedDateTime`

### PR DESCRIPTION
**Pull request by omus**
Since `ZonedDateTime` parsing code heavily relies on `Dates` from the standard library I originally did not write tests for this logic. I have now added a few tests to ensure the basics work and additionally have:

* Added `istimezone` function which can determine if a string is valid for `TimeZone`
* Added a `default_format(::Type{ZonedDateTime})` method to extend `Dates.default_format`
* Cleaned up a fallback function for `parse_components`

### CODE REVIEW

**Comment by omus** [comment]
cc: @iamed2

File: test/parse.jl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        [
+            Dates.Year(2017),
+            Dates.Month(11),
+            Dates.Day(14),
+            Dates.Hour(11),
+            Dates.Minute(3),
+            Dates.Second(53),
+            tz"UTC+01",
+        ]
+    )
```
Comment (iamed2) [inline test/parse.jl]: The formatting of this block makes me sad

Comment (omus) [reply test/parse.jl]: lol, that is fair. I don't love it myself. I think I may just save the expected value to a variable first. Any suggestions?

Comment (iamed2) [reply test/parse.jl]: That, and not using `==` in function form. That's probably it.

**Comment by omus** [comment]
Should be fine to merge as I just refactored the `==()` test. I expect 0.7 to currently fail.

**Pull request merged**